ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -579,8 +579,7 @@ public static Map<String, Object> buildStreamsProperties(
         ? queryId.get().toString()
         : QueryApplicationId.PERSISTENT_QUERY_INDICATOR;
 
-    if (config.getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)
-        && type.equals(QueryApplicationId.PERSISTENT_QUERY_INDICATOR)) {
+    if (!queryId.isPresent()) {
```

Comment:
This is super confusing, what is the `queryId` supposed to represent? 

IIUC we basically only want to do this if we're building a persistent, and gen 2 query -- it's not at all clear that this is what we're doing here though, I mean why would `queryId` not be present for only that one specific case?

Also I wonder if we can/should get rid of the `type` thing that you had introduced in the other PR when we fixed this for transient topics?

Reply:
The query Id is to represent the query id. Since this is building a runtime if there is no queryid it is a shared runtime

Reply:
Yes we can get rid of the type